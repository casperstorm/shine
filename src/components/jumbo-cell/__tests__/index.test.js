import renderer from 'react-test-renderer'
import React from 'react'

import JumboCell from '../index'

describe('jumbo-cell', () => {
  const title = 'foo'
  const description = 'bar'

  it('should render correctly with all props', () => {
    const tree = renderer.create(
      <JumboCell title={title} description={description} onPress={() => {}} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly without props', () => {
    const tree = renderer.create(<JumboCell />)
    expect(tree).toMatchSnapshot()
  })
})
