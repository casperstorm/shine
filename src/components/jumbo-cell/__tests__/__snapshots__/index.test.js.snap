// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jumbo-cell should render correctly with all props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 15,
          "marginLeft": 16,
          "marginTop": 59,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={1}
        />
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Rubik-Bold",
          "fontSize": 30,
          "lineHeight": 38,
          "marginLeft": 12,
          "marginRight": 12,
          "textAlign": "left",
        }
      }
    >
      foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Rubik-Light",
          "fontSize": 15,
          "marginBottom": 25,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 10,
          "opacity": 0.7,
        }
      }
    >
      bar
    </Text>
    <View
      style={
        Object {
          "borderTopColor": "#444444",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "left": 12,
          "opacity": 0.8,
          "position": "absolute",
          "right": 12,
        }
      }
    />
  </View>
</View>
`;

exports[`jumbo-cell should render correctly without props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginLeft": 16,
        "marginRight": 16,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 15,
          "marginLeft": 16,
          "marginTop": 59,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={1}
        />
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Rubik-Bold",
          "fontSize": 30,
          "lineHeight": 38,
          "marginLeft": 12,
          "marginRight": 12,
          "textAlign": "left",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Rubik-Light",
          "fontSize": 15,
          "marginBottom": 25,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 10,
          "opacity": 0.7,
        }
      }
    />
    <View
      style={
        Object {
          "borderTopColor": "#444444",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "left": 12,
          "opacity": 0.8,
          "position": "absolute",
          "right": 12,
        }
      }
    />
  </View>
</View>
`;
